File: app/src/main/java/com/example/wuganjizhang/data/local/converter/ListConverter.java
package com.example.wuganjizhang.data.local.converter;

import androidx.room.TypeConverter;

public class ListConverter {
    
    @TypeConverter
    public static String fromList(java.util.List<String> list) {
        if (list == null || list.isEmpty()) {
            return null;
        }
        return android.text.TextUtils.join(",", list);
    }

    @TypeConverter
    public static java.util.List<String> toList(String value) {
        if (value == null || value.isEmpty()) {
            return java.util.Collections.emptyList();
        }
        return java.util.Arrays.asList(value.split(","));
    }
}

File: app/src/main/java/com/seamless/bookkeeper/data/local/converter/ListConverter.kt
package com.seamless.bookkeeper.data.local.converter

import androidx.room.TypeConverter

class ListConverter {
    @TypeConverter
    fun fromStringList(value: String?): List<String>? = value?.split(",")?.map { it.trim() }

    @TypeConverter
    fun toStringList(list: List<String>?): String? = list?.joinToString(",")

    @TypeConverter
    fun fromLongList(value: String?): List<Long>? = value?.split(",")?.mapNotNull { it.trim().toLongOrNull() }

    @TypeConverter
    fun toLongList(list: List<Long>?): String? = list?.joinToString(",")
}
